tasks/preload.js

```diff
@@ -109,28 +110,45 @@ module.exports = function (grunt) {
 		}
 
 		['components', 'libraries'].forEach(function(preloadType) {
+			var preloadInfo, preloadOptions, aVersionParts;
 
-			var preloadOptions = preloadData[preloadType];
-
+			preloadOptions = preloadData[preloadType];
 			if (!preloadOptions) {
 				return;
 			}
 
-			var preloadInfo = (preloadType === 'libraries') ? {
-				moduleName: 'library-preload',
-				ext: '.json',
-				indicatorFile: 'library.js',
-				processModuleName: function(moduleName) {
-					return moduleName.replace(/\//g, '.');
-				}
-			} : {
-				moduleName: 'Component-preload',
-				ext: '.js',
-				indicatorFile: 'Component.js',
-				processContent: function(content) {
-					return 'jQuery.sap.registerPreloadedModules(' + content + ');';
+			if (preloadType === 'libraries') {
+				preloadInfo = {
+					moduleName: 'library-preload',
+					indicatorFile: 'library.js',
+					processModuleName: function(moduleName) {
+						return moduleName.replace(/\//g, '.');
+					}
+				};
+
+				aVersionParts = options.compatVersion.split(".");
```

Comment:
`compatVersion` should be validated with a proper error message in case of invalid input.